modules/internal/ChapelBase.chpl

```diff
@@ -636,17 +636,70 @@ module ChapelBase {
   //
   // left and right shift on primitive types
   //
-  inline proc <<(a: int(?w), b: integral) return __primitive("<<", a, b);
-  inline proc <<(a: uint(?w), b: integral) return __primitive("<<", a, b);
 
-  inline proc >>(a: int(?w), b: integral) return __primitive(">>", a, b);
-  inline proc >>(a: uint(?w), b: integral) return __primitive(">>", a, b);
+  inline proc bitshiftChecks(a, b: integral) {
+    const hdr = "Cannot bitshift " + a:string + " by " + b:string +
+                " because ";
```

Comment:
Will this result in constructing this string (piece by piece) for every shift operation (when bounds checking is on)?  That seems expensive.

Reply:
You're right. Perhaps I should just go with a less constructive error message?